narwhals/_pandas_like/dataframe.py

```diff
@@ -165,7 +167,8 @@ def select(
         if not new_series:
             # return empty dataframe, like Polars does
             return self._from_dataframe(self._dataframe.__class__())
-        new_series = validate_indices(new_series)
+        if self._implementation != "dask":
+            new_series = validate_indices(new_series)
```

Comment:
rather than this, could we just always validate indices, and then, if indices aren't aligned, then for dask, we raise an error, whereas for other implementations we use `set_axis` and continue?